### PR TITLE
once conflict review feature is unmerged w/ undo reintroduce merge button

### PR DESCRIPTION
**Pull request by jackgrossman18**
ref #1635

### CODE REVIEW

**Comment by maxgrossman** [comment]
This worked for sure when I clicked merge, then undo back to before all the actions within a merge (change tags, delete).

However, if I use undo/redo to go to the head and base of the undo/redo stack (click undo or redo until I cannot anymore in either direction), the button gets all out of sync.

Perhaps we can listen to the undone/redone events fired by the history class?
The class is accessible on context so it'd be avail in this class.

I think you'd be able to say something like...

on `undone`, if the thing that is being undone (via the annotation on the graph) is changed tags, we know we can show the merge button. This is because the first step of the merge is to change the tags of the target feature.

on `redone` If the thing being redone is that then you'd be able to say ok hide the merge button. (since that's the inverse of what we care about for undone)

**Review by maxgrossman — CHANGES_REQUESTED** [review]
Nice! I added a comment. This seems to work in one direction (when I click undone).
We'll need this to be bi-directional (undoing and redoing), and I added an idea of how we might be able to achieve it.

**Comment by brianhatchl** [comment]
@bobi7 I made the changes suggested by @maxgrossman.  Rather than checking for what change the unde/redo is peforming, it seems sufficient to just check the review features and make sure they are eligible for merge.  The test at play here is that both review features exist in the graph, which is not true after merge, but becomes true again when the merge is undone.

**Comment by brianhatchl** [comment]
should also fix #1006 #1480

**Comment by brianhatchl** [comment]
might also add a fix for #1476

File: modules/Hoot/ui/conflicts/info.js
Comment (brianhatchl) [inline modules/Hoot/ui/conflicts/info.js:47]: @jackgrossman18 because this method listens to history changes, it will be called outside the conflation review context and so getting the current relation may throw an "entity not found" error.

File: modules/Hoot/ui/conflicts/info.js
Comment (brianhatchl) [inline modules/Hoot/ui/conflicts/info.js:53]: @jackgrossman18 this seems to me a simpler way to handle if a feature has been merged and 1) had tags updated or 2) been deleted.  By grabbing the features from the graph instead of the stale `currentFeatures` the tag table will now update correctly anytime review merges happen.

File: modules/Hoot/ui/conflicts/map.js
Comment (brianhatchl) [inline modules/Hoot/ui/conflicts/map.js:61]: @jackgrossman18 move this logic inside `merge.js`

File: modules/Hoot/ui/conflicts/merge.js
Comment (brianhatchl) [inline modules/Hoot/ui/conflicts/merge.js:46]: @jackgrossman18 listen for undo/redo and call the same function that is called when a new review is loaded.

File: modules/Hoot/ui/conflicts/merge.js
Comment (brianhatchl) [inline modules/Hoot/ui/conflicts/merge.js:57]: @jackgrossman18 we have to deep clone here because we modify the feature tags on line 68.  If we do a merge, then undo it and reverse it the element merge service errors because both featuers have the `hoot:merge:target=yes` tag.

Comment (brianhatchl) [reply modules/Hoot/ui/conflicts/merge.js:57]: Also when checking if reverse merge is valid, we disallow if it is POI->Poly.  We can simply disallow because code in `checkMergeButton` ;omr 358 makes sure the poi and poly features are always in the right order.

File: modules/Hoot/ui/conflicts/merge.js
Comment (brianhatchl) [inline modules/Hoot/ui/conflicts/merge.js:325]: I found that the original code was putting the arrow direction in the opposite from the intended from/to direction

File: modules/Hoot/ui/conflicts/merge.js
Comment (brianhatchl) [inline modules/Hoot/ui/conflicts/merge.js:354]: Use dynamic features retrieved from graph.  This allows the show/hide merge button to work on undo/redo.

File: modules/Hoot/ui/conflicts/merge.js
Comment (brianhatchl) [inline modules/Hoot/ui/conflicts/merge.js:359]: We need to define POI->Poly merge this way (instead of using `hoot:review:type` because after a merge, other review relations will be updated to refer to the merged element.  After a poi->poly merge the referred element may no long be a POI and may be two polys but the review type remains unchanged.  This may change when we implement this long ago ticket #1003 to allow poly->poly merges.

File: modules/Hoot/ui/conflicts/merge.js
Comment (brianhatchl) [inline modules/Hoot/ui/conflicts/merge.js:368]: Make sure the feature order for the merge array is always poi->poly

File: modules/Hoot/ui/conflicts/merge.js
Comment (bobi7) [inline modules/Hoot/ui/conflicts/merge.js]: should we wrap the `d3.event.ctrlKey || d3.event.metaKey` in parenthesis for readability?

**Comment by brianhatchl** [comment]
Also fixes #1214

**Comment by brianhatchl** [comment]
and fixes #1795

**Review by bobi7 — APPROVED** [review]

**Pull request merged**
